client/DESIGN.md

```diff
@@ -0,0 +1,292 @@
+# Treode Client Library Interface
+
+The two major components of the Treode client library interface are the [Cache](#Cache) and the [Transaction](#Transaction).  
+
+Caches store key-value pairs retrieved from the server, as well as the read and value timestamps associated with each pair that Treode provides.  Caches are responsible for maintaining a connection to the DB server.  Users may create multiple caches to work on disjoint sections of the database simultaneously.  
+
```

Comment:
> Users may create multiple caches

This is true, but I think it's distracting to bring that point into the conversation.